File: 0125-valid-palindrome/solution.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        a=""
        s=s.lower()
        for i in s:
            if((i>='a' and i<='z') or (i>='A' and i<='Z' or i>='0' and i<='9')):
                a=a+i
        print(a)
        if(a==a[::-1]):
            return True
        else:
            return False
        

File: 0125-valid-palindrome/solution.cpp
class Solution {
public:

    bool palin(int i,string &s){
        int n=s.length();
        if(i>=n/2) return true;

        if(s[i]!=s[n-i-1]) return false;
        return palin(i+1,s);

    }

    bool isPalindrome(string s) {
        
        string ans;
        for(char c:s){
            if(isalnum(c))
            ans+=tolower(c);
        }
        return palin(0,ans);
        

    }
};
